### ISSUE TITLE
Image does not display in SimpleDraweeView when open react-native page for the first time

### ISSUE THREAD

**Issue by elyonsaber** [facebook/fresco#1683]
I wrapped up android native UI component, and integrated it with my react native application. The layout of this component includes SimpleDraweeView 
`

    <com.facebook.drawee.view.SimpleDraweeView
        android:id="@+id/avatar_image"
        android:layout_width="26dp"
        android:layout_height="26dp"
        android:layout_marginTop="6dp"
        android:layout_centerHorizontal="true"
        android:scaleType="centerCrop"
        fresco:failureImageScaleType="fitCenter"
        fresco:placeholderImage="@drawable/gift_box_0"
        fresco:placeholderImageScaleType="centerCrop"
        fresco:roundAsCircle="true" />
</RelativeLayout>`
And I set imageUri of SimpleDraweeView dynamically in code:

`SimpleDraweeView avatar = (SimpleDraweeView) infoWindow.findViewById(R.id.avatar_image);
             avatar.setImageURI(UserRepository.getInstance().getUserInfo().getAvatar().getUrl());`

However, when I first open the react page, it only shows placeholder image. If I go back to the previous page, and enter this react page again, then the drawee displays the correct image of uri. So do I miss any setting? Or is this the issue of React Native? Thanks!

**Comment by kirwan** [comment]
Nothing looks obviously wrong from what you've detailed here but I'd be interested where you're setting the URI.

You could compare what you have with the [ReactImageView](https://github.com/facebook/react-native/blob/e1577df1fd70049ce7f288f91f6e2b18d512ff4d/ReactAndroid/src/main/java/com/facebook/react/views/image/ReactImageView.java) class in React Native. Particularly, I'd suggest looking at where they call `maybeUpdateView()` which is where they initiate the request.

Beyond that, you might have more luck asking this question on the React Native project as I don't have much experience of working with that.

**Comment by elyonsaber** [comment]
Thanks for you prompt reply! Just as you mentioned, I thought the problem was caused by that I set the URI, and then add the view to map as a marker, like this:
```
View infoWindow = layoutInflater.inflate(R.layout.custom_map_location, LL, false);
SimpleDraweeView avatar = (SimpleDraweeView) infoWindow.findViewById(R.id.avatar_image);
avatar.setImageURI(UserRepository.getInstance().getUserInfo().getAvatar().getUrl());
Marker marker = map.addMarker(
new MarkerOptions().position(new LatLng(amapLocation.getLatitude(),amapLocation.getLongitude()))
                           .anchor(0.5f, 1f)
                           .icon(BitmapDescriptorFactory.fromView(infoWindow)));
```
I should have waited util the loading of image was complete before add it to map as a marker, so I try to set control listener to SimpleDraweeView:
```
 mControllerListener = new BaseControllerListener<ImageInfo>() {
    @Override
    public void onFinalImageSet(String id,
                       @Nullable final ImageInfo imageInfo,
                       @Nullable Animatable animatable) {
         Marker marker = map.addMarker(
              new MarkerOptions().position(new LatLng(amapLocation.getLatitude(), ampLocation.getLongitude()))        
                                 .anchor(0.5f, 1f)// 锚点设置
                                 .icon(BitmapDescriptorFactory.fromView(infoWindow)));
                                }

     @Override
     public void onFailure(String id, Throwable throwable) {
             LogUtil.d(TAG, "on failure");
     }
  };

 DraweeController controller = Fresco.newDraweeControllerBuilder()
                                    .setUri(Uri.parse(UserRepository.getInstance().getUserInfo().getAvatar().getUrl()))
                                    .setControllerListener(mControllerListener)
                                    .setOldController(avatar.getController())
                                    .build();

 avatar.setController(controller);
``` 
But this time I found that neither onFinalImageSet nor onFailure was called. 
So could you plz give me any more suggestion? Great thanks!

**Comment by elyonsaber** [comment]
After several tries, I finally found a way to solve this issue. 
SimpleDraweeView called onAttach DraweeHolder in onAttachedToWindow() and onStartTemporaryDetach() , so if I just inflate a layout and don't add it to root view to display it, these two methods seems not to be called. 
I wrote a custom view extends imageView according to [docs](https://www.fresco-cn.org/docs/writing-custom-views.html). In the custom view, I call draweeHolder.onAttach in constructor, and call`super.setImageDrawable(mDraweeHolder.getTopLevelDrawable());` to display drawable, otherwise the Drawee won't appear at all.
